project/tests/test_cross_spark_publish.py

```diff
@@ -0,0 +1,364 @@
+#!/usr/bin/env python3
+"""
+Cross-Spark Version Build Testing
+
+Tests the Delta Lake build system by validating JAR file names for:
+1. Default publish (publishM2) - should publish ALL modules
+2. Spark-specific publish (runOnlyForSparkModules) - should publish only Spark-dependent modules
+
+Usage:
+    python project/tests/test_cross_spark_publish.py
+
+The script will:
+1. Test default publishM2 command publishes all modules for default Spark version
+2. Test runOnlyForSparkModules command publishes only Spark-dependent modules
+3. Test full cross-version build workflow
+4. Exit with status 0 on success, 1 on failure
+"""
+
+import subprocess
+import sys
+from dataclasses import dataclass
+from pathlib import Path
+from typing import List, Set, Dict
+
+
+# Spark-related modules (requiresCrossSparkBuild := true)
+# These modules get a Spark version suffix (e.g., _4.0) for non-default versions
+# Template format: {suffix} = Spark version suffix (e.g., "", "_4.0")
+#                  {version} = Delta version (e.g., "3.4.0-SNAPSHOT")
+SPARK_RELATED_JAR_TEMPLATES = [
+    "delta-spark{suffix}_2.13-{version}.jar",
+    "delta-connect-common{suffix}_2.13-{version}.jar",
+    "delta-connect-client{suffix}_2.13-{version}.jar",
+    "delta-connect-server{suffix}_2.13-{version}.jar",
+    "delta-sharing-spark{suffix}_2.13-{version}.jar",
+    "delta-contribs{suffix}_2.13-{version}.jar",
+    "delta-iceberg{suffix}_2.13-{version}.jar"
+]
+
+# Non-spark-related modules (built once, same for all Spark versions)
+# Template format: {version} = Delta version (e.g., "3.4.0-SNAPSHOT")
+NON_SPARK_RELATED_JAR_TEMPLATES = [
+    # Scala modules
+    "delta-hudi_2.13-{version}.jar",
```

Comment:
do we need to cross-spark publish this as well?

Reply:
cc @lzlfred